src/test/java/frc/robot/filters/TestAutoYawFilter.java

```diff
@@ -0,0 +1,61 @@
+package frc.robot.filters;
+
+import org.junit.jupiter.api.Test;
+
+import edu.wpi.first.wpilibj.Timer;
+
+import static org.junit.Assert.assertEquals;
+
+import frc.utils.TestGyro;
+
+public class TestAutoYawFilter {
+    private static final double SMALLDIFFERENCE = 0.01;
+    
+    @Test
+    public void testAutoYawA() {
+        Timer timer = new Timer();
+        TestGyro gyro = new TestGyro(90, 7.5);
+        AutoYawFilter filter = new AutoYawFilter(gyro);
+        DriveInput DI = new DriveInput(0, 1, 0);
+
+        filter.setEnabled(true);
+        timer.start();
+
+        while (!(gyro.getAngle() < 1 && gyro.getAngle() > -1)) {
```

Comment:
this is pretty confusing, since TestGyro is counting strictly upwards.  I feel like this can be simplified